site/src/api/queries/users.ts

```diff
@@ -1,5 +1,13 @@
+import { QueryClient } from "@tanstack/react-query";
 import * as API from "api/api";
-import { UpdateUserPasswordRequest } from "api/typesGenerated";
+import { UpdateUserPasswordRequest, UsersRequest } from "api/typesGenerated";
+
+export const users = (req: UsersRequest) => {
+  return {
+    queryKey: ["users", req],
```

Comment:
should we really be using `req` as a key here? not `req.id` or something?

Reply:
Req is basically the `page`, `offset` and `filter` params so in this case I think makes sense to use them together. Reference: https://tanstack.com/query/v4/docs/react/guides/query-keys#query-keys-are-hashed-deterministically

Reply:
Yeah, it's not always needed, but I've seen the pattern where the payload for the query is treated as the last argument of the query key. It also makes it possible for the query function to access it via the `queryKey` property on its options argument, even if you extract the query function outside the hook